theano/tensor/basic.py

```diff
@@ -1825,12 +1825,39 @@ def neq(a, b):
 def isnan(a):
     """isnan(a)"""
 
+# Rename isnan to isnan_ to allow to bypass it when not needed.
+# glibc 2.23 don't allow isnan on int, so we remove it from the graph.
+isnan_ = isnan
+
+
+def isnan(a):
+    """isnan(a)"""
+    a = as_tensor_variable(a)
+    if a.dtype in discrete_dtypes:
+        return alloc(numpy.asarray(False, dtype="bool"),
+                     *[a.shape[i] for i in range(a.ndim)])
+    return isnan_(a)
+
 
 @_scal_elemwise
 def isinf(a):
     """isinf(a)"""
 
 
+# Rename isnan to isnan_ to allow to bypass it when not needed.
+# glibc 2.23 don't allow isnan on int, so we remove it from the graph.
```

Comment:
I'm pretty sure nobody cares about which version of glibc accepts or doesn't accept this.

Also this has probably more to do with the compiler than the libc since the prototypes exist for float and double and it's up to the compiler to upcast other types.